3rdParty/iresearch/core/index/index_writer.cpp

```diff
@@ -878,7 +888,19 @@ index_writer::flush_context_ptr index_writer::documents_context::update_segment(
   return ctx;
 }
 
-void index_writer::flush_context::emplace(active_segment_context&& segment) {
+void index_writer::flush_context::AddToPending(
+    active_segment_context& segment) {
+  if (segment.flush_ctx_) {
```

Comment:
Should this check be executed under the mutex as well? Otherwise it seems there is a potential for a data race.

Reply:
mutex_ guard pending_segment_contexts_

Reply:
`segment` is owned exclusively by thread. So no need to sync.